crates/e2e/tests/e2e/ethflow.rs

```diff
@@ -121,6 +121,20 @@ async fn eth_flow_tx_zero_fee(web3: Web3) {
 
     test_order_was_settled(&services, &ethflow_order, &web3).await;
 
+    // make sure the fee was charged for zero fee limit orders
+    let fee_charged = || async {
+        onchain.mint_block().await;
+        let order = services
+            .get_order(&ethflow_order.uid(onchain.contracts()).await)
+            .await
+            .unwrap();
+        match order.metadata.class {
```

Comment:
The test should not implement this match statement but instead know if the order is supposed to be a limit order (and thus ave a non zero fee or not). Let's kill whatever of the two branches is not applicable here.

Reply:
fixed